integrationTests/chainSimulator/vm/esdtImprovements_test.go

```diff
@@ -1932,3 +1932,238 @@ func TestChainSimulator_SFT_ChangeMetaData(t *testing.T) {
+	tx := issueTx(0, addrs[0].Bytes, funTicker, baseIssuingCost)
+
+	txResult, err := cs.SendTxAndGenerateBlockTilTxIsExecuted(tx, maxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, txResult)
+	require.Equal(t, "success", txResult.Status.String())
+	funTokenID := txResult.Logs.Events[0].Topics[0]
+
+	log.Info("Issued FungibleESDT token id", "tokenID", string(funTokenID))
+
+	metaData := txsFee.GetDefaultMetaData()
+	metaData.Nonce = []byte(hex.EncodeToString(big.NewInt(1).Bytes()))
+
+	createTokenUpdateTokenIDAndTransfer(t, cs, addrs[0].Bytes, addrs[1].Bytes, funTokenID, metaData, epochForDynamicNFT, addrs[0])
+
+	shardID := cs.GetNodeHandler(0).GetProcessComponents().ShardCoordinator().ComputeId(addrs[0].Bytes)
+
+	checkMetaData(t, cs, core.SystemAccountAddress, funTokenID, shardID, metaData)
+	checkMetaDataNotInAcc(t, cs, addrs[0].Bytes, funTokenID, shardID)
+	checkMetaDataNotInAcc(t, cs, addrs[1].Bytes, funTokenID, shardID)
+}
+
+func TestChainSimulator_MetaESDTCreatedBeforeSaveToSystemAccountEnabled(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	baseIssuingCost := "1000"
+	cs, epochForDynamicNFT := getTestChainSimulatorWithSaveToSystemAccountDisabled(t, baseIssuingCost)
+	defer cs.Close()
+
+	addrs := createAddresses(t, cs, false)
+
+	log.Info("Initial setup: Create MetaESDT that will have it's metadata saved to the user account")
+
+	metaTicker := []byte("METATICKER")
+	tx := issueMetaESDTTx(0, addrs[0].Bytes, metaTicker, baseIssuingCost)
+
+	txResult, err := cs.SendTxAndGenerateBlockTilTxIsExecuted(tx, maxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, txResult)
+	require.Equal(t, "success", txResult.Status.String())
+	metaTokenID := txResult.Logs.Events[0].Topics[0]
+
+	log.Info("Issued MetaESDT token id", "tokenID", string(metaTokenID))
+
+	metaData := txsFee.GetDefaultMetaData()
+	metaData.Nonce = []byte(hex.EncodeToString(big.NewInt(1).Bytes()))
+
+	createTokenUpdateTokenIDAndTransfer(t, cs, addrs[0].Bytes, addrs[1].Bytes, metaTokenID, metaData, epochForDynamicNFT, addrs[0])
+
+	shardID := cs.GetNodeHandler(0).GetProcessComponents().ShardCoordinator().ComputeId(addrs[0].Bytes)
+	checkMetaData(t, cs, core.SystemAccountAddress, metaTokenID, shardID, metaData)
+	checkMetaDataNotInAcc(t, cs, addrs[0].Bytes, metaTokenID, shardID)
+	checkMetaDataNotInAcc(t, cs, addrs[1].Bytes, metaTokenID, shardID)
+}
+
+func getTestChainSimulatorWithSaveToSystemAccountDisabled(t *testing.T, baseIssuingCost string) (testsChainSimulator.ChainSimulator, int32) {
+	startTime := time.Now().Unix()
+	roundDurationInMillis := uint64(6000)
+	roundsPerEpoch := core.OptionalUint64{
+		HasValue: true,
+		Value:    20,
+	}
+
+	activationEpochForSaveToSystemAccount := uint32(2)
+	activationEpochForDynamicNFT := uint32(4)
+
+	numOfShards := uint32(3)
+	cs, err := chainSimulator.NewChainSimulator(chainSimulator.ArgsChainSimulator{
+		BypassTxSignatureCheck:   false,
```

Comment:
we need to set `BypassTxSignatureCheck` to `true` after last merged changes from here https://github.com/multiversx/mx-chain-go/pull/6268

Reply:
Done